File: src/test/java/com/ragin/bdd/cucumbertests/hooks/SimpleCustomUUIDMatcher.java
package com.ragin.bdd.cucumbertests.hooks;

import com.ragin.bdd.cucumber.matcher.BddCucumberJsonMatcher;
import org.hamcrest.BaseMatcher;
import org.hamcrest.Description;
import org.springframework.lang.NonNull;

public class SimpleCustomUUIDMatcher extends BaseMatcher<String> implements BddCucumberJsonMatcher {
    @Override
    public String matcherName() {
        return "isUUID";
    }

    @Override
    public Class<? extends BaseMatcher<?>> matcherClass() {
        return this.getClass();
    }

    @Override
    public boolean matches(final @NonNull Object actual) {
        if (actual instanceof String) {
            String actualString = String.valueOf(actual);
            return actualString.matches("[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}");
        }
        return false;
    }

    @Override
    public void describeTo(Description description) {
    }
}

File: bdd-cucumber-gherkin-lib/src/test/kotlin/com/ragin/bdd/cucumbertests/hooks/SimpleCustomUUIDMatcher.kt
package com.ragin.bdd.cucumbertests.hooks

import com.ragin.bdd.cucumber.matcher.BddCucumberJsonMatcher
import org.hamcrest.BaseMatcher
import org.hamcrest.Description

class SimpleCustomUUIDMatcher : BaseMatcher<String>(), BddCucumberJsonMatcher {
    override fun matcherName(): String {
        return "isUUID"
    }

    override fun matcherClass(): Class<out BaseMatcher<*>?> {
        return this.javaClass
    }

    override fun matches(actual: Any): Boolean {
        if (actual is String) {
            return actual
                .matches("[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}".toRegex())
        }
        return false
    }

    override fun describeTo(description: Description) = Unit
}
